server/src/test/java/org/elasticsearch/common/time/DateFormattersTests.java

```diff
@@ -1371,4 +1371,11 @@ public void testParsingMissingTimezone() {
         long millisJoda = DateFormatter.forPattern("yyyy-MM-dd HH:mm:ss").parseMillis("2018-02-18 17:47:17");
         assertThat(millisJava, is(millisJoda));
     }
+
+    // see https://bugs.openjdk.org/browse/JDK-8193877
+    public void testNoClassCastException() {
+        String input = "DpNKOGqhjZ";
+        IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> DateFormatter.forPattern(input));
+        assertThat(e.getMessage(), containsString(input));
```

Comment:
Let's also check the cause, to make sure we actually got the ClassCastException we expected

Reply:
The only additional information we have here is the exception message, I'll add a check for that.

Reply:
The ClassCastException which was caught by your change is added as the cause to the created IllegalArgumentException. We should be able to check the cause for that?

Reply:
Yes sorry, I misinterpreted your first comment, I thought you meant to check that the CCE was actually due to the JDK bug, then I realized you just suggested one more assert in the test to check that the `IllegalArgumentException`'s root cause was a CCE.
I fixed it with my latest commit.